File: factory/code-typescript/deuxiemeMethode/fabriqueConcretes/ProduitFactory2.ts
import { Produit } from "../Produit.js";
import { ProduitFactory } from "../ProduitFactory.js";
import { Produit2 } from "../produitsConcrets/Produit2.js";

export class ProduitFactory2 extends ProduitFactory {
  constructor() {
    super();
  }

  createProduit(): Produit {
    return new Produit2();
  }
}

File: 1-patrons-de-creations/factory/code-java/deuxiemeMethode/fabrique_concretes/ProduitFactory2.java
package fabrique_concretes;

import produit_factory.ProduitFactory;
import produit.Produit;
import produits_concrets.Produit2;

public class ProduitFactory2 extends ProduitFactory {
    protected Produit createProduit(){
        return new Produit2();
    }
}